### PR TITLE
[Fix] relay onnx frontend bug when [A, B, M, N] * [1, B, N, K]

### PR DESCRIPTION
**Pull request by willzhang4a58**
Thanks for contributing to TVM!   Please refer to guideline https://tvm.apache.org/docs/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/apache/incubator-tvm/blob/master/CONTRIBUTORS.md#reviewers) by @ them in the pull request thread.

### CODE REVIEW

**Comment by willzhang4a58** [comment]
Fix BUG and add test case for batch_matmul [A, B, M, N] * [1, B, N, K] 

cc @Huyuwei @jwfromm @kazum @mbrookhart @siju-samuel

**Comment by AndrewZhaoLuo** [comment]
Will take a look later today

File: python/tvm/relay/frontend/onnx.py [hunk 1]
```diff
@@ -268,6 +256,40 @@ def flatten_to_nd(x, x_shape, nd=3):
...
+            a = flatten_to_nd(inputs[0], a_shape, 2)
+            b = _op.transpose(inputs[1])
+            output = _op.nn.dense(a, b, out_dtype=out_dtype)
+        else:
+            # broadcast a and b
+            a_broadcasted_shape = _op.concatenate(
+                [
+                    out_batch,
+                    _op.strided_slice(
+                        a_shape, [infer_shape(a_shape)[0] - 2], [infer_shape(a_shape)[0]]
```
Comment (AndrewZhaoLuo) [inline python/tvm/relay/frontend/onnx.py:272]: nit: use `a_rank` instead of `infer_shape(a_shape)[0]`

**Review by AndrewZhaoLuo — APPROVED** [review]
LGTM, just please address nits

**Comment by willzhang4a58** [comment]
hi @AndrewZhaoLuo , I have fixed the comment.

**Pull request merged**